data/transactions/logic/eval_test.go

```diff
@@ -123,23 +129,27 @@ func benchmarkEvalParams(txn *transactions.SignedTxn) *EvalParams {
 	return ep
 }
 
-func defaultEvalParamsWithVersion(txn *transactions.SignedTxn, version uint64) *EvalParams {
-	var zero uint64
-	ep := &EvalParams{
-		Proto:     makeTestProtoV(version),
-		TxnGroup:  make([]transactions.SignedTxnWithAD, 1),
-		Specials:  &transactions.SpecialAddresses{},
-		Trace:     &strings.Builder{},
-		FeeCredit: &zero,
-		SigLedger: MakeLedger(nil),
+func defaultEvalParamsWithVersion(version uint64, txns ...transactions.SignedTxn) *EvalParams {
+	empty := false
+	if len(txns) == 0 {
+		empty = true
+		txns = []transactions.SignedTxn{{Txn: transactions.Transaction{Type: protocol.ApplicationCallTx}}}
 	}
-	if txn != nil {
-		ep.TxnGroup[0].SignedTxn = *txn
+	ep := NewEvalParams(transactions.WrapSignedTxnsWithAD(txns), makeTestProtoV(version), &transactions.SpecialAddresses{})
+	ep.Trace = &strings.Builder{}
+	ep.SigLedger = NewLedger(nil)
+	if empty {
+		// We made an app type in order to get a full ep, but that sets MinTealVersion=2
+		ep.TxnGroup[0].Txn.Type = "" // set it back
+		ep.MinAvmVersion = nil       // will recalculate in eval()
 	}
-	ep.reset()
 	return ep
 }
 
+func (ep *EvalParams) isFull() bool {
```

Comment:
Is this a right name? My understanding is one might have `available` set and it does not mean nothing can be added

Reply:
I'm using "Full" here to mean "A complete EvalParams, capable of evaluating apps".

I'm willing to change the name, but this is test code, not even "public" to normal code in package.

Reply:
@algorandskiy I attempted to address the feedback by selecting a new name in https://github.com/algorand/go-algorand/pull/4692.  Can you review https://github.com/algorand/go-algorand/pull/4692?

I'll resolve the thread once the child PR is merged.